src/main/java/us/talabrek/ultimateskyblock/uSkyBlock.java

```diff
@@ -1082,13 +1100,17 @@ public boolean createIsland(final Player player, final PlayerInfo pi) {
                     playerInfo.setIslandGenerating(false);
                 }
 
+                // Clear first, since the player could log out and we NEED to make sure their inventory gets cleared.
+                clearPlayerInventory(player);
+                clearEntitiesNearPlayer(player);

```

Comment:
If you are using "multiverse inventories" for instance, you will have separate inventories in separate worlds. The above change you introduced, will do the clearing before teleporting the player, meaning their inventory will be lost, if they invoke the command from the "wrong world", or, they will be able to exploit this bug, so they keep their items from the skyworld, if they do it from another world.